share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```
@@ -198,7 +201,12 @@ function(ExternalZephyrProject_Add)
     # Only set image specific board if provided.
     # The sysbuild BOARD is exported through sysbuild cache, and will be used
     # unless <image>_BOARD is defined.
-    list(APPEND sysbuild_cache_strings "${ZBUILD_APPLICATION}_BOARD:STRING=${ZBUILD_BOARD}\n")
+    if(DEFINED ZBUILD_BOARD_REVISION)
+      # Use provided board revision
+      list(APPEND sysbuild_cache_strings "${ZBUILD_APPLICATION}_BOARD:STRING=${ZBUILD_BOARD}@${ZBUILD_BOARD_REVISION}\n")
+    else()
+      list(APPEND sysbuild_cache_strings "${ZBUILD_APPLICATION}_BOARD:STRING=${ZBUILD_BOARD}\n")
+    endif()
```

we need a check that revision is not set without `BOARD` being set also so we can inform the caller instead of silently just ignore `REVISION`.

Ignoring an argument can easily lead to confusion later.

For example:
```suggestion
    endif()
  elseif(DEFINED ZBUILD_BOARD_REVISION)
    message(FATAL_ERROR "ExternalZephyrProject_Add(... BOARD_REVISION ${ZBUILD_BOARD_REVISION}) requires BOARD.")
```

---

Fixed.